### PR TITLE
tests: Add extra tests to check the case when a rule return nil action

### PR DESCRIPTION
**Pull request by infrmtcs**

### CODE REVIEW

File: consensus/tendermint/rule_proposal_and_polka_current_test.go [hunk 1]
```diff
@@ -32,6 +32,43 @@ func TestProposalAndPolkaCurrent(t *testing.T) {
 		assertState(t, stateMachine, types.Height(0), types.Round(0), types.StepPrecommit)
 	})
 
+	t.Run("Line 36: lock and broadcast precommit when step is precommit", func(t *testing.T) {
```
Comment (rianhughes) [inline consensus/tendermint/rule_proposal_and_polka_current_test.go:35]: Why did you add this test? I'm all for it. But it seems like you were concerned that line 36 would block line 49?

**Review by rianhughes — APPROVED** [review]

File: consensus/tendermint/rule_proposal_and_polka_current_test.go (hunk 1 above)
Comment (infrmtcs) [reply consensus/tendermint/rule_proposal_and_polka_current_test.go:35]: Yes, that's correct. I want to make sure that if line 36 return nil action, line 49 can still trigger.
